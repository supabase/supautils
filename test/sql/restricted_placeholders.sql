-- placeholder non-restricted values pass
select set_config('response.headers', '[{"Cache-Control": "public"}, {"Cache-Control": "max-age=259200"}]', true);

-- placeholder restricted values fail on a set_config
select set_config('response.headers', '[{"Content-Type": "application/vnd.my.malicious.type"}, {"Cache-Control": "public"}]', true);
select set_config('response.headers', '[{"X-Special-Header": "a-value"}]', true);
\echo

-- placeholder restricted values fail on a SET
set another.placeholder to 'special-value';
\echo

-- doesn't crash after a show all
\o /dev/null
show all;
\o
select 1;
