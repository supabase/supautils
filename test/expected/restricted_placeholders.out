-- placeholder non-restricted values pass
select set_config('response.headers', '[{"Cache-Control": "public"}, {"Cache-Control": "max-age=259200"}]', true);
                             set_config                             
--------------------------------------------------------------------
 [{"Cache-Control": "public"}, {"Cache-Control": "max-age=259200"}]
(1 row)

-- placeholder restricted values fail on a set_config
select set_config('response.headers', '[{"Content-Type": "application/vnd.my.malicious.type"}, {"Cache-Control": "public"}]', true);
ERROR:  The placeholder contains the "content-type" disallowed value
select set_config('response.headers', '[{"X-Special-Header": "a-value"}]', true);
ERROR:  The placeholder contains the "x-special-header" disallowed value
\echo

-- placeholder restricted values fail on a SET
set another.placeholder to 'special-value';
ERROR:  The placeholder contains the "special-value" disallowed value
\echo

-- doesn't crash after a show all
\o /dev/null
show all;
\o
select 1;
 ?column? 
----------
        1
(1 row)

